### PR TITLE
Implement `Bytes` as one of the type for Class, Instance and Group Identifiers

### LINKED ISSUE #105: Introduce `tagged bytes` as one of the types within Environment Identifiers (ClassID, Instance and Group)
With the introduction of `tagged bytes` as one of the type for Class, Instance and Group Identifiers in CoRIM Draft
(see https://github.com/ietf-rats-wg/draft-ietf-rats-corim/commit/7af90a6577564336571b4d8b6c585137ed21c02d)

this issue tracks the implementation of this modification.

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**
Fixes #105

### CODE REVIEW

File: comid/group.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/veraison/corim/extensions"
 )
 
-// Group stores a group identity.  The supported format is UUID.
+// Group stores a group identity. The supported format is UUID and a variable length opaque bytes.
```
Comment (thomas-fossati) [inline comid/group.go:15]:
```suggestion
// Group stores a group identity. The supported formats are UUID and variable-length opaque bytes.
```

File: comid/group.go [hunk 2]
```diff
@@ -53,10 +70,18 @@ func (o *Group) UnmarshalCBOR(data []byte) error {
...
+//		  "type": "uuid",
+//		  "value": "69E027B2-7157-4758-BCB4-D9F167FE49EA"
+//		}
+//
+// (b) Tagged bytes, e.g. :
 //
 //	{
-//	  "type": "uuid",
+//	  "type": "bytes",
 //	  "value": "69E027B2-7157-4758-BCB4-D9F167FE49EA"
```
Comment (thomas-fossati) [inline comid/group.go:85]: this doesn't look right (UUID format)

File: comid/group_test.go [hunk 3]
```diff
@@ -60,3 +67,125 @@ func TestGroup_UmarshalJSON(t *testing.T) {
...
+}
+
+func TestGroup_MarshalJSONBytes_OK(t *testing.T) {
+	testBytes := []byte{0x01, 0x02, 0x03}
+	tb := TaggedBytes(testBytes)
+	tv := Group{&tb}
+	jsonBytes, err := tv.MarshalJSON()
+	require.NoError(t, err)
+	assert.Equal(t, `{"type":"bytes","value":"AQID"}`, string(jsonBytes))
+
```
Comment (thomas-fossati) [inline comid/group_test.go:143]: spurious NL
```suggestion
```

File: comid/instance.go [hunk 4]
```diff
@@ -77,6 +77,7 @@ func (o *Instance) UnmarshalCBOR(data []byte) error {
 //
 //	ueid: base64-encoded bytes, e.g. "YWNtZS1pbXBsZW1lbnRhdGlvbi1pZC0wMDAwMDAwMDE="
 //	uuid: standard UUID string representation, e.g. "550e8400-e29b-41d4-a716-446655440000"
+//	bytes: a variable length opaque bytes, example {0x07, 0x12, 0x34}
```
Comment (thomas-fossati) [inline comid/instance.go:80]:
```suggestion
//	bytes: a variable-length opaque byte string, example {0x07, 0x12, 0x34}
```

File: comid/instance_test.go [hunk 5]
```diff
@@ -67,3 +68,125 @@ func Test_RegisterInstanceType(t *testing.T) {
...
+	}
+}
+
+func Test_NewBytesInstance_NOK(t *testing.T) {
+	for _, tv := range []struct {
+		Name  string
+		Input any
+		Err   string
+	}{
+
```
Comment (thomas-fossati) [inline comid/instance_test.go:93]: spurious NL
```suggestion
```

File: comid/instance_test.go [hunk 6]
```diff
@@ -67,3 +68,125 @@ func Test_RegisterInstanceType(t *testing.T) {
...
+		Name  string
+		Input string
+	}{
+		{
+			Name:  "valid input test 1",
+			Input: `{ "type": "bytes", "value": "MTIzNDU2Nzg5" }`,
+		},
+		{
+			Name:  "valid input test 2",
+			Input: `{ "type": "bytes", "value": "deadbeef"}`,
```
Comment (thomas-fossati) [inline comid/instance_test.go:158]: this example is slightly confusing as it gives the impression that base-16 is a valid format :-)

**Review by thomas-fossati — COMMENTED** [review]
There's some repetition that could be possibly avoided in the tests.

Please have a look at the linter suggestions (not those about the heavy parameters, the others) and fix them.  They are simple things that remove extra typography.

Also, I have left a couple of editorial fixes.

Other than that LGTM, thanks!

**Comment by thomas-fossati** [comment]
you've probably missed some linters warnings, but I see you have already hit the merge button so it's going to be next time

**Comment by yogeshbdeshpande** [comment]
Its just one as I see in `instance_test.go` I will fix it in more wider `lint` clean up under the Issue #91

**Pull request merged**